pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -542,7 +547,7 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				status.UpdatedReplicas++
 			}
 			// if the set does not allow bursting, return immediately
-			if monotonic || decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
+			if monotonic {
 				return &status, nil
 			}
```

Comment:
```
if monotonic {
    return &status, nil
} else if decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
    break
}

if isTerminating(replicas[i]) && monotonic {
	klog.V(4).Infof(
		"StatefulSet %s/%s is waiting for Pod %s to Terminate",
		set.Namespace,
		set.Name,
		replicas[i].Name)
	return &status, nil
} else if isTerminating(replicas[i]) && decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
     break
}

if monotonic || scaleMaxUnavailable != nil {
	isAvailable, waitTime := isRunningAndAvailable(replicas[i], minReadySeconds)
	if !isAvailable && monotonic {
		if waitTime > 0 {
			// make sure we check later
			durationStore.Push(getStatefulSetKey(set), waitTime)
			klog.V(4).Infof(
				"StatefulSet %s/%s needs to wait %s for the Pod %s to be Running and Available after being"+
					" Ready for %d seconds",
				set.Namespace,
				set.Name,
				waitTime,
				replicas[i].Name,
				minReadySeconds)
		} else {
			klog.V(4).Infof(
				"StatefulSet %s/%s is waiting for Pod %s to be Running and Ready",
				set.Namespace,
				set.Name,
				replicas[i].Name)
		}
		return &status, nil
	} else if !isAvailable && decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
		break
	}
}
```

Reply:
If it is convenient for you, you can communicate on the zoom of the community meeting on 12.8.